pkg/scheduler/plugins/gpushare/device_handler.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gpushare
+
+import (
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+func (n *nodeDeviceCache) onDeviceAdd(obj interface{}) {
+	device, ok := obj.(*schedulingv1alpha1.Device)
+	if !ok {
+		return
+	}
+	n.assign(device.Name, device)
```

Comment:
assign -> update may be better, assign is too specific concept like "assign pod"